dvc/dependency/base.py

```diff
@@ -21,9 +23,11 @@ def __init__(self, path):
 class BaseDependency:
     IS_DEPENDENCY = True
 
-    DoesNotExistError = DependencyDoesNotExistError
-    IsNotFileOrDirError = DependencyIsNotFileOrDirError
-    IsStageFileError = DependencyIsStageFileError
+    DoesNotExistError = DependencyDoesNotExistError  # type: Type[DvcException]
```

Comment:
Found a way to override subtypes. :)

Reply:
So, no need for a hack/refactor for now. cc @efiop.

Reply:
That's hacky :D But will do until we refactor outs/deps.